python/ray/autoscaler/updater.py

```diff
@@ -144,7 +145,31 @@ def do_update(self):
         self.provider.set_node_tags(
             self.node_id, {TAG_RAY_NODE_STATUS: "SettingUp"})
         for cmd in self.setup_cmds:
-            self.ssh_cmd(cmd, verbose=True)
+            deadline = time.time() + NODE_START_WAIT_S
```

Comment:
this retry works - only for the NodeUpdater in charge of the headnode though (since ray is installed from pip)

Reply:
Why do this? The timeout can just be increased for this one?

Reply:
Hm this is to address the following error:
```
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
```
IIRC the command fails even before hitting the timeout. The suggestion Robert shared was to modify some config file and reboot the instance. Are there other workarounds for this?

Reply:
I guess one could do this:
```
while sudo fuser /var/lib/dpkg/lock >/dev/null 2>&1; do
    sleep 1
done

Reply:
Yeah, let's do it in the command instead of the in the python